CI Workflow: `create-review-issues` Fails with GitHub CLI Syntax Error

<details>
<summary>Original Issue Body</summary>

Run pnpm tsx scripts/ci/run-create-review-issues.ts
🔍 Fetching existing issues to prevent duplicates...
🛡️ Verifying required labels exist...
✨ Found missing labels: bot-generated, triage-needed, type-technical-debt, priority-low. Creating them...
   - Label "bot-generated" already exists (likely created by a parallel process).
   - Label "triage-needed" already exists (likely created by a parallel process).
   - Label "type-technical-debt" already exists (likely created by a parallel process).
   - Label "priority-low" already exists (likely created by a parallel process).
🚀 Creating issue: "Refine SpotifyPolling SDK Access with `!` Operator"...
Unhandled error: Error: GitHub CLI Error: /bin/sh: 1: Syntax error: end of file unexpected

    at GitHubClient.execute (/home/runner/work/hrm/hrm/scripts/create-review-issues.ts:116:13)
    at GitHubClient.createIssue (/home/runner/work/hrm/hrm/scripts/create-review-issues.ts:203:24)
    at run (/home/runner/work/hrm/hrm/scripts/create-review-issues.ts:335:12)
    at <anonymous> (/home/runner/work/hrm/hrm/scripts/ci/run-create-review-issues.ts:8:1)
    at ModuleJob.run (node:internal/modules/esm/module_job:325:25)
    at async ModuleLoader.import (node:internal/modules/esm/loader:606:24)
    at async asyncRunEntryPointWithESMLoader (node:internal/modules/run_main:117:5)
Error: Process completed with exit code 1.

</details>

The automated workflow for creating review issues (`scripts/ci/run-create-review-issues.ts`) is currently failing with a syntax error from the GitHub CLI. This prevents the system from automatically tracking technical debt and review items.

**Error Log:**
```
Run pnpm tsx scripts/ci/run-create-review-issues.ts
🔍 Fetching existing issues to prevent duplicates...
🛡️ Verifying required labels exist...
✨ Found missing labels: bot-generated, triage-needed, type-technical-debt, priority-low. Creating them...
   - Label "bot-generated" already exists (likely created by a parallel process).
   - Label "triage-needed" already exists (likely created by a parallel process).
   - Label "type-technical-debt" already exists (likely created by a parallel process).
   - Label "priority-low" already exists (likely created by a parallel process).
🚀 Creating issue: "Refine SpotifyPolling SDK Access with `!` Operator"...
Unhandled error: Error: GitHub CLI Error: /bin/sh: 1: Syntax error: end of file unexpected

    at GitHubClient.execute (/home/runner/work/hrm/hrm/scripts/create-review-issues.ts:116:13)
    at GitHubClient.createIssue (/home/runner/work/hrm/hrm/scripts/create-review-issues.ts:203:24)
    at run (/home/runner/work/hrm/hrm/scripts/create-review-issues.ts:335:12)
    at <anonymous> (/home/runner/work/hrm/hrm/scripts/ci/run-create-review-issues.ts:8:1)
    at ModuleJob.run (node:internal/modules/esm/module_job:325:25)
    at async ModuleLoader.import (node:internal/modules/esm/loader:606:24)
    at async asyncRunEntryPointWithESMLoader (node:internal/modules/run_main:117:5)
Error: Process completed with exit code 1.
```

**Context & Potential Cause:**
The stack trace indicates the error originates during the `GitHubClient.createIssue` call at `/home/runner/work/hrm/hrm/scripts/create-review-issues.ts:203:24`. The `Syntax error: end of file unexpected` from `/bin/sh` strongly suggests that the command arguments being passed to the `gh` CLI for issue creation are malformed. The issue title being processed at the time, "Refine SpotifyPolling SDK Access with `!` Operator", contains characters (backticks, exclamation mark) that require careful escaping or quoting in a shell context. This likely points to an issue with how the `GitHubClient` constructs or escapes the arguments for the `gh issue create` command.

--- a/scripts/create-review-issues.ts
+++ b/scripts/create-review-issues.ts
@@ -199,9 +199,11 @@ export class GitHubClient implements IGitHubClient {
       writeFileSync(bodyFile, body, 'utf-8')
 
       // FIX: The --title-file flag is not valid. Use --title with the title string directly.
-      // To prevent shell injection issues with complex titles, escape double quotes.
-      const escapedTitle = title.replace(/"/g, '\\"')
-      const cmd = `gh issue create --title "${escapedTitle}" --body-file "${bodyFile}" --label "${labels}"`
+      // To prevent shell injection issues with complex titles, escape single quotes
+      // and wrap the title in single quotes for the shell. This is a robust way
+      // to handle special characters like '!', '$', '`', etc.
+      const escapedTitle = title.replace(/'/g, "'\\''")
+      const cmd = `gh issue create --title '${escapedTitle}' --body-file "${bodyFile}" --label "${labels}"`
       const url = this.execute(cmd)
       console.log(`✅ Issue created: ${url}`)
     } finally {

--- a/utils/socketManager.ts
+++ b/utils/socketManager.ts
@@ -33,7 +33,6 @@ import { roundTo } from '../lib/utils.js'
 // Define service instances to be managed
 // New: Define a function to get the state snapshot
 let getUnifiedStateSnapshot: () => StateSnapshot
-// Store WebSocket server reference for command relay
 let wsServerInstance: WebSocketServer
 let connectionMonitor: ConnectionMonitor
 let services: AppServices

